machine-runner/src/runner/runner.ts

```diff
@@ -409,26 +423,34 @@ export const createMachineRunnerInternal = <
               })
 
               // Effects of handlingReport on emitters
-              ;(() => {
-                if (pushEventResult.executionHappened) {
-                  if (emitter.listenerCount('audit.state') > 0) {
-                    emitter.emit('audit.state', {
-                      state: ImplStateOpaque.make<SwarmProtocolName, MachineName, StateUnion>(
-                        internals,
-                        internals.current,
-                      ),
-                      events: pushEventResult.triggeringEvents,
-                    })
-                  }
-                }
-
-                if (!pushEventResult.executionHappened && pushEventResult.discardable) {
-                  emitter.emit('audit.dropped', {
-                    state: internals.current.data,
-                    event: pushEventResult.discardable,
+              if (pushEventResult.type === PushEventTypes.React) {
+                if (emitter.listenerCount('audit.state') > 0) {
+                  emitter.emit('audit.state', {
```

Comment:
out of curiosity: what happens with exceptions thrown by event listeners?

Reply:
That has never occurred to me. 

It throws the `emitter.emit` call as well.
Severely, if an emitter has several listeners and one in the middle throws, only the previous listeners are executed.
This is something we cannot control

Reply:
We can do something like this though
```
export class ErrorIgnoringEventEmitter extends EventEmitter {
  emit(eventName: string | symbol, ...args: any[]): boolean {
    const listeners = this.listeners(eventName)

    listeners.forEach((listener) => {
      try {
        listener(...args)
      } catch (error) {}
    })

    return listeners.length > 0
  }
}
```

Reply:
Then we’d also have to handle `once`, right? Perhaps a reasonable compromise is to catch and log anything that escapes `emitter.emit`.

Reply:
> Then we’d also have to handle once, right? Perhaps a reasonable compromise is to catch and log anything that escapes emitter.emit.

I tried `const listeners = this.listeners(eventName)` and it handles `once` automatically. It seems `once` wraps the listener with an unsubscribe in the end

Reply:
then all is good